### ISSUE TITLE
perf: add linux perftool recipes for investigating IO perf

### ISSUE THREAD

**Issue by petermattis** [cockroachdb/pebble#724]
We should figure out the appropriate [linux perftools](https://github.com/brendangregg/perf-tools) incantations to trace IO syscalls. I believe we can hook at the block layer and capture stacktraces all the way up into the Pebble code. 

Related, we should write up some recipes for using [blktrace](https://linux.die.net/man/8/blktrace).

**Comment by petermattis** [comment]
For `blktrace`, see https://tunnelix.com/debugging-disk-issues-with-blktrace-blkparse-btrace-and-btt-in-linux-environment/.

**Comment by petermattis** [comment]
@itsbilal You were using `blktrace` recently. Can you add your some brief usage notes here? I'll take on the task of figuring out a more permanent location in the repo to hold our knowledge.

**Comment by itsbilal** [comment]
@petermattis Ah, missed this last week. Basically I found this resource helpful: https://brooker.co.za/blog/2013/07/14/io-performance.html

At first, I ran `blktrace -w 30 -d /dev/nvme1n1 -o sdd` to inspect IO events on `/dev/nvme1n1` (where the first EBS volume usually goes), for 30 seconds, with the output going to `sdd` in the current directory.

Then, I ran `blkparse -i ./sdd | grep RS | grep cockroach` to see all synchronous reads (RS) events issued by the cockroach process.

After that point I was able to manually step through the output and figure out what was happening (I also had a "better" output file from a run w/ rocksdb to compare, which helped a lot). But had that not worked, I'd have looked into using `btt` or additional tools to understand the blktrace output better.
